akka-docs/src/main/paradox/typed/cluster.md

```diff
@@ -95,15 +98,327 @@ Scala
+as unreachable and removed after the automatic or manual downing.
+1. A more graceful exit can be performed if you tell the cluster that a node shall leave. 
+This can be performed using @ref:[JMX](../additional/operations.md#jmx) or @ref:[HTTP](../additional/operations.md#http).
+ 
+The @ref:[Coordinated Shutdown](../actors.md#coordinated-shutdown) will automatically run when the cluster node sees itself as
+`Exiting`, i.e. leaving from another node will trigger the shutdown process on the leaving node.
+Tasks for graceful leaving of cluster including graceful shutdown of Cluster Singletons and
+Cluster Sharding are added automatically when Akka Cluster is used, i.e. running the shutdown
+process will also trigger the graceful leaving if it's not already in progress.
+
+Normally this is handled automatically, but in case of network failures during this process it might still
+be necessary to set the node’s status to `Down` in order to complete the removal.
+
+## Serialization 
+ 
+Enable @ref:[serialization](../serialization.md) to send events between ActorSystems and systems
+external to the Cluster. @ref:[Serialization with Jackson](../serialization-jackson.md) is a good choice in many cases, and our
+recommendation if you don't have other preferences or constraints.
+ 
+Actor references are typically included in the messages, since there is no `sender`. 
+To serialize actor references to/from string representation you would use the `ActorRefResolver`.
+For example here's how a serializer could look for `Ping` and `Pong` messages:
 
 Scala
 :  @@snip [PingSerializer.scala](/akka-cluster-typed/src/test/scala/docs/akka/cluster/typed/PingSerializer.scala) { #serializer }
 
 Java
 :  @@snip [PingSerializerExampleTest.java](/akka-cluster-typed/src/test/java/jdocs/akka/cluster/typed/PingSerializerExampleTest.java) { #serializer }
+
+You can look at the
+@java[@extref[Cluster example project](samples:akka-samples-cluster-java)]
+@scala[@extref[Cluster example project](samples:akka-samples-cluster-scala)]
+to see what this looks like in practice.
+
+## Node Roles
+
+Not all nodes of a cluster need to perform the same function: there might be one sub-set which runs the web front-end,
+one which runs the data access layer and one for the number-crunching. Deployment of actors—for example by cluster-aware
+routers—can take node roles into account to achieve this distribution of responsibilities.
+
+The node roles are defined in the configuration property named `akka.cluster.roles`
+and typically defined in the start script as a system property or environment variable.
+
+The roles are part of the membership information in `MemberEvent` that you can subscribe to.
+
+## Failure Detector
+
+The nodes in the cluster monitor each other by sending heartbeats to detect if a node is
+unreachable from the rest of the cluster and has been downed and removed. Please see:
```

Comment:
The heartbeating is not about being downed and removed, only about being unreachable, downing and removing could be a consequence (if using SBR)

Reply:
Thats directly from https://doc.akka.io/docs/akka/current/common/cluster.html#failure-detector

Reply:
What we are trying to say here is not that heartbeating as about downing or removed, but an underlying mechanism of detection, which feeds back into things like...

Reply:
Yep, which is why I think the phrasing `and has been downed and removed` is not good

Reply:
I'm on the same page now, have removed that in two files to just:
`The nodes in the cluster monitor each other by sending heartbeats to detect if a node is
unreachable from the rest of the cluster.`